controllers/sms-gateway.js

```diff
@@ -119,6 +120,7 @@ module.exports = {
       .then(JSON.parse)
       .then(function(request) {
         // Process webapp-terminating messages asynchronously
+        // TODO these promises do nothing!
```

Comment:
The point of the promises was to make the processing of messages happen after the response has been sent to the client, and the connection closed.  I appreciate they may not have been doing this.